sdk/src/test/scala/com/horizen/account/validation/BaseFeeBlockValidatorTest.scala

```diff
@@ -0,0 +1,49 @@
+package com.horizen.account.validation
+
+import com.horizen.account.FeeUtils
+import com.horizen.account.block.AccountBlock
+import com.horizen.account.history.AccountHistory
+import com.horizen.account.utils.AccountMockDataHelper
+import org.junit.Assert.assertTrue
+import org.junit.Test
+import org.scalatestplus.junit.JUnitSuite
+
+import java.math.BigInteger
+import scala.compat.java8.OptionConverters.RichOptionForJava8
+
+class BaseFeeBlockValidatorTest extends JUnitSuite {
+
+  @Test
+  def nonGenesisBlockCheck(): Unit = {
+    var mockHelper: AccountMockDataHelper = AccountMockDataHelper(true)
+    val mockedGenesisBlock = mockHelper.getMockedBlock(FeeUtils.INITIAL_BASE_FEE)
+    var mockedBlock: AccountBlock = mockHelper.getMockedBlock(BigInteger.ZERO)
+    var mockedHistory: AccountHistory = mockHelper.getMockedAccountHistory(Option.apply(mockedGenesisBlock).asJava)
+
+    // Test 1: Successful validation, block is genesis block with initial base fee
+    assertTrue(BaseFeeBlockValidator().validate(mockedGenesisBlock, mockedHistory).isSuccess)
+
+    // Test 2: Validation exception expected, block is genesis block and does not have initial base fee
+    assertThrows[InvalidBaseFeeException] {
+      BaseFeeBlockValidator().validate(mockedBlock, mockedHistory).get
+    }
+
+    mockHelper = AccountMockDataHelper(false)
+    mockedHistory = mockHelper.getMockedAccountHistory(Option.apply(mockedGenesisBlock).asJava)
+    mockedBlock = mockHelper.getMockedBlock(FeeUtils.INITIAL_BASE_FEE)
+
+    // Test 3: Successful validation, block is empty, therefore base fee did not change
```

Comment:
This comment is a bit misleading. You construct two blocks with the same base fee, so validation should succeed.
If these were real blocks the base should decrease by 12.5% after an empty block, right?

Reply:
Yes, you are right - I mean gasUsed is the same, therefore same baseFee. Just saw that the other comment below is false, too. 🙈